### PR TITLE
#372 | Broadcast mode in subscription

### PR DESCRIPTION
**Pull request by sbolimowski**

### CODE REVIEW

File: hermes-console/static/partials/subscription.html [hunk 1]
```diff
@@ -111,6 +111,9 @@ <h3 class="panel-title">Properties</h3>
                         <strong>Delivery type:</strong> {{subscription.deliveryType}}
                         <span uib-popover='Hermes can deliver messages in SERIAL (one message at a time) or in BATCH (group of messages at a time).' popover-trigger="mouseenter" class="fa helpme pull-right">&#xf128;</span>
                     </p>
+                    <p>
+                        <strong>Mode:</strong> {{subscription.mode}}
```
Comment (adamdubiel) [inline hermes-console/static/partials/subscription.html]: can we add a popover with explaination of mode?

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java [hunk 2]
```diff
@@ -131,8 +133,12 @@ private boolean shouldReduceSendingRate(MessageSendingResult result) {
         return !result.isRetryLater() && subscriptionAllowsResending(result);
     }
 
+    private boolean messageSentSucceeded(MessageSendingResult result) {
+        return result.succeeded() || (result.isClientError() && !subscription.getSerialSubscriptionPolicy().isRetryClientErrors());
+    }
+
     private boolean subscriptionAllowsResending(MessageSendingResult result) {
```
Comment (adamdubiel) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java]: i dont quite get the naming here: so subscription allows resending if sending failed?

File: hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/Message.java [hunk 3]
```diff
@@ -82,8 +86,9 @@ public boolean isTtlExceeded(long ttlMillis) {
         return currentTimestamp > readingTimestamp + ttlMillis;
     }
 
-    public void incrementRetryCounter() {
+    public void incrementRetryCounter(Collection<String> succeededUris) {
         this.retryCounter++;
+        this.succeededUris = ImmutableSet.copyOf(succeededUris);
```
Comment (adamdubiel) [inline hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/Message.java]: shouldn't this be an append operation? and i don't know if we want to create new collection for each retry, think about our poor allocation rate :) maybe appending to the set would be enough, sacrificing the internal immutability?

File: hermes-console/static/partials/subscription.html (hunk 1 above)
Comment (sbolimowski) [inline hermes-console/static/partials/subscription.html]: done

**Comment by adamdubiel** [comment]
👍

**Comment by wojtkiewicz** [comment]
👍

**Pull request merged**